### PR TITLE
Add com.wickeditor.WickEditor

### PR DESCRIPTION
**Pull request by Aman9das**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** The project is largely not under development anymore.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Aman9das** [comment]
bot, build com.wickeditor.WickEditor

File: com.wickeditor.WickEditor.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+build-options:
+  append-path: /usr/lib/sdk/node14/bin
+  cflags: -O2 -g
+  cxxflags: -O2 -g
```
Comment (hfiguiere) [inline com.wickeditor.WickEditor.yml:19]:
```suggestion
```

**Review by barthalion — CHANGES_REQUESTED** [review]
`shared-modules` were added directly instead of git submodule.

**Comment by barthalion** [comment]
Have you reached out to upstream developers about your submission?

File: com.wickeditor.WickEditor.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    buildsystem: simple
+    build-commands:
+      - install -D com.wickeditor.WickEditor.png /app/share/icons/hicolor/128x128/apps/com.wickeditor.WickEditor.png
+      - install -D com.wickeditor.WickEditor.desktop /app/share/applications/com.wickeditor.WickEditor.desktop
+      - install -D com.wickeditor.WickEditor.metainfo.xml /app/share/metainfo/com.wickeditor.WickEditor.metainfo.xml
+    sources:
+      - type: file
+        path: screenshot.png
+      - type: file
+        path: com.wickeditor.WickEditor.png
```
Comment (hfiguiere) [inline com.wickeditor.WickEditor.yml:65]: there is an icon file at https://github.com/Wicklets/wick-editor/blob/master/public/icon.png

why do you need this one?

File: com.wickeditor.WickEditor.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        . ../flatpak-node/electron-builder-arch-args.sh
+        npm run package-linux -- $ELECTRON_BUILDER_ARCH_ARGS  --linux --dir
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+  - name: WickEditor-misc
+    buildsystem: simple
+    build-commands:
+      - install -D com.wickeditor.WickEditor.png /app/share/icons/hicolor/128x128/apps/com.wickeditor.WickEditor.png
```
Comment (hfiguiere) [inline com.wickeditor.WickEditor.yml:58]: this icon is NOT 128x128 but rather 521x512

**Review by hfiguiere — CHANGES_REQUESTED** [review]
You should submit the .desktop and the metainfo.xml upstream.

The lack of .desktop upstream is a bug.

**Comment by Aman9das** [comment]
bot, build com.wickeditor.WickEditor

File: com.wickeditor.WickEditor.yml (hunk 3 above)
Comment (hfiguiere) [reply com.wickeditor.WickEditor.yml:58]: and the metainfo and a properly sized icons.

**Comment by Aman9das** [comment]
Hello! May I ask what all are holding up the publishing of wick editor

**Comment by hfiguiere** [comment]
patience?

**Comment by Aman9das** [comment]
> patience?

Thanks! My bad.. i thought i something something was needed at my end

**Pull request closed without merging**